Sources/MapboxDirections/TollCost.swift

```diff
@@ -0,0 +1,190 @@
+import Foundation
+import Turf
+
+/**
+ :nodoc:
+ Toll cost information for the `Route`.
+ */
+public struct TollCost: Equatable,  Codable, ForeignMemberContainer {
+    public var foreignMembers: Turf.JSONObject = [:]
+    
+    private enum CodingKeys: String, CodingKey {
+        case currency
+        case paymentMethods = "payment_methods"
+    }
+
+    /**
+     Related currency string.
+     
+     Uses ISO 4217 format. Refers to values in `CostPerVehicleSize`.
+     */
+    public let currency: String
+    /**
+     Information about toll payment methods.
+     */
+    public let paymentMethods: PaymentMethods
+    
+    init(currency: String, paymentMethods: PaymentMethods) {
+        self.currency = currency
+        self.paymentMethods = paymentMethods
+    }
+        
+    /**
+     :nodoc:
+     Information about toll payment methods.
+     */
+    public struct PaymentMethods: Equatable, Codable, ForeignMemberContainer {
+        public var foreignMembers: Turf.JSONObject = [:]
+        
+        private enum CodingKeys: String, CodingKey {
+            case ETC = "etc"
+            case cash
+        }
+        
+        /**
+         Information about payment by ETC.
+         */
+        public let ETC: CostPerVehicleSize?
+        /**
+         Information about payment by cash.
+         */
+        public let cash: CostPerVehicleSize?
+        
+        init(ETC: CostPerVehicleSize? = nil, cash: CostPerVehicleSize? = nil) {
+            self.ETC = ETC
+            self.cash = cash
+        }
+    }
+    
+    /**
+     :nodoc:
+     `PaymentMethods` details about particular vehicle size.
+     
+     Vehicle sizes are [standartized](https://en.wikipedia.org/wiki/Expressways_of_Japan#Tolls).
+     */
+    public struct CostPerVehicleSize: Equatable, Codable, ForeignMemberContainer {
+        public var foreignMembers: Turf.JSONObject = [:]
+        
+        private enum CodingKeys: String, CodingKey {
+            case small
+            case standard
+            case middle
+            case large
+            case jumbo
+        }
+        
+        /**
+         The toll cost for a small sized vehicle.
+         
+         A toll cost of `0` is valid and simply means that no toll costs are incurred for this route.
+         */
+        public let small: Double?
```

Comment:
Frameworks such as [Intents](https://developer.apple.com/documentation/sirikit/inridefarelineitem/1649903-price) and [StoreKit](https://developer.apple.com/documentation/storekit/product/3749586-price) type `price` as a `Decimal`, probably because if you do enough arithmetic on prices represented as floating-point numbers, [rounding error starts to add up](https://stackoverflow.com/a/3730040/4585461). It’s unlikely to be a problem in practice when calculating the price of tolls along a single route, but we should be consistent with software that does have to worry about this issue. Let’s try using `Decimal` in our own representation and convert back to `Double` when encoding as JSON.